src/core/validations/cast_test.rs

```diff
@@ -0,0 +1,98 @@
+mod tests {
+    use rand::Rng;
+    use serde::Deserialize;
+
+    use crate::{
+        core::validations,
+        proto::{cast_add_body, embed, Embed},
+    };
+
+    #[derive(Deserialize)]
+    struct Message {
+        data: MessageData,
+    }
+
+    #[derive(Deserialize)]
+    struct MessageData {
+        #[serde(rename = "castAddBody")]
+        cast_add_body: CastAddBody,
+    }
+
+    #[derive(Deserialize)]
+    struct CastAddBody {
+        #[serde(rename = "embedsDeprecated")]
+        embeds_deprecated: Vec<String>,
+        mentions: Vec<u64>,
+        #[serde(rename = "parentCastId")]
+        parent_cast_id: Option<ParentCastId>,
+        text: String,
+        #[serde(rename = "mentionsPositions")]
+        mentions_positions: Vec<u64>,
+        embeds: Vec<String>,
+        #[serde(rename = "type")]
+        cast_type: String,
+    }
+
+    #[derive(Deserialize)]
+    struct ParentCastId {
+        fid: u64,
+        hash: String,
+    }
+
+    #[derive(Deserialize)]
+    struct PagedResponse {
+        messages: Vec<Message>,
+    }
+
+    #[tokio::test]
+    async fn test_cast_validation() {
+        let n: u32 = rand::thread_rng().gen::<u32>() % 10000;
+        let resp = reqwest::get(format!(
+            "https://hoyt.farcaster.xyz:2281/v1/castsByFid?fid={}",
```

Comment:
This might end up being flaky. If it is, I'd add this as a separate test that's run manually and CI uses a known set of messages checked into the repo.